### PR TITLE
Customizable faster random

### PR DESCRIPTION
**Pull request by HaHaWTH**
This pull request includes following changes:
- Detailed error for MultithreadedTracker
- Faster random generator for Entity random
- Option for enabling L64X128MixRandomSource for worldgen
- Allocate 1 thread for AsyncLocator by default

Tested and shouldn't break any farms/machines.

### CODE REVIEW

File: patches/server/0126-Multithreaded-Tracker.patch [hunk 1]
```diff
@@ -435,7 +435,9 @@ index 0000000000000000000000000000000000000000..1ecaa9f4b4843c05af2139ae3e4b0dd0
 +                tickAsyncWithCompatMode(level);
 +            }
 +        } catch (Exception e) {
-+            LOGGER.error("Failed to execute async task.", e);
++            LOGGER.error("Error occurred while executing async task.");
++            //noinspection CallToPrintStackTrace
++            e.printStackTrace();
```
Comment (Dreeam-qwq) [inline patches/server/0126-Multithreaded-Tracker.patch]: Original method can also print stacktrace. Is there a reason to use printStackTrace() ?

Comment (HaHaWTH) [reply patches/server/0126-Multithreaded-Tracker.patch]: Done. uwu

**Comment by HaHaWTH** [comment]
New changes:
- Better SmoothTeleport API by checking world height internally to fix client visual issues and network protocol error.
- Deprecated the old api and marked as for removal.

**Review by Dreeam-qwq — APPROVED** [review]

**Comment by Dreeam-qwq** [comment]
Crab Crab

**Pull request merged**